docs/resources/cluster_agent_token.md

```diff
@@ -0,0 +1,66 @@
+# Resource: cluster_agent_token
+
+This resource allows you to create and manage cluster agent tokens.
+
+Buildkite Documentation: https://buildkite.com/docs/clusters/manage-clusters#set-up-clusters-connect-agents-to-a-cluster
+
+## Example Usage
+
+```hcl
+provider "buildkite" {}
+
+resource "buildkite_cluster_agent_token" "cluster-token-1" {
+  cluster_id = "Q2x1c3Rlci0tLTkyMmVjOTA4LWRmNWItNDhhYS1hMThjLTczMzE0YjQ1ZGYyMA==" 
+  description = "agent token for cluster-1" 
+}
+```
+
+## Argument Reference
+
+* `cluster_id` - (Required) The ID of the cluster that this cluster queue belongs to.
+* `description` -  (Required) A description about what this cluster agent token is used for.
+
+## Attribute Reference
+
+* `id` - The Graphql ID of the created cluster queue.
+* `uuid` - The UUID of the created cluster queue.
+* `cluster_uuid` - The UUID of the cluster that this cluster queue belongs to.
+
+
+To find the cluster's `UUID` to utilize, you can use the below GraphQL query below. Alternatively, you can use this [pre-saved query](https://buildkite.com/user/graphql/console/3adf0389-02bd-45ef-adcd-4e8e5ae57f25), where you will need fo fill in the organization slug (ORGANIZATION_SLUG) for obtaining the relevant cluster name/`UUID` that the cluster queue is in.
```

Comment:
I think we actually need the ID for this resource right? The UUID is a computed property

Reply:
This new resource needs the `cluster_id` to link to a cluster, but this example query demonstrates how to get the `cluster_uuid`. So we need to update the wording and query to retrieve `cluster_id` instead since that is what is required